### ISSUE TITLE
Fast gradient Method doesn't work for double precision: Suggestion for the fix

### ISSUE THREAD

**Issue by williampiat3** [cleverhans-lab/cleverhans#1224]
***The issue tracker should only be used to report bugs or feature requests. If you are looking for support from other library users, please ask a question on [StackOverflow](https://stackoverflow.com/questions/tagged/cleverhans).***

**Describe the bug**
I used the PGD method in this folder but I use double precision in my models thus encountered an error while using the attack 
cleverhans/cleverhans/torch/attacks/projected_gradient_descent.py

I found a way of fixing the problem:
 line 74 in cleverhans/cleverhans/torch/attacks/fast_gradient_method.py
instead of:
```python
x = x.clone().detach().to(torch.float).requires_grad_(True)
```
I put
```python
x = x.clone().detach().to(x.dtype).requires_grad_(True)
```
My solution then covers double precision and simple precision

**Comment by kylematoba** [comment]
this also breaks for `torch.float16`.
